mismi-s3/src/Mismi/S3/Commands.hs

```diff
@@ -453,25 +453,25 @@ foldWR e a = \case
   WorkerErr err -> e err
 
 
-retryAWSAction :: RetryPolicy -> AWS a -> AWS a
-retryAWSAction rp a = do
-  local (retryAWS' rp) $ a
+retryAWSAction' :: Retry -> AWS a -> AWS a
+retryAWSAction' r =
+  local (override (serviceRetry .~ r))
 
-retryAWS :: Int -> Env -> Env
-retryAWS i = retryAWS' (retryWithBackoff i)
-
-retryAWS' :: RetryPolicy -> Env -> Env
-retryAWS' _ e =
-  let err c v = case v of
-        NoResponseDataReceived -> True
-        StatusCodeException s _ _ -> s == status500
-        FailedConnectionException _ _ -> True
-        FailedConnectionException2 _ _ _ _ -> True
-        TlsException _ -> True
-        _ -> (e ^. envRetryCheck) c v
-  in
-  e & envRetryCheck .~ err
+retryAWSAction :: Int -> AWS a -> AWS a
+retryAWSAction =
+  local . retryAWS
 
+retryAWS :: Int -> Env -> Env
+retryAWS i e = e & envRetryCheck .~ err
```

Comment:
I don't think there is a way to specify it now. Although `amazonka` does use [exponential approach](https://github.com/ambiata/amazonka/blob/ambiata/amazonka-s3/gen/Network/AWS/S3/Types.hs#L482-L488) internally.